File: ruby/testcode/loginjection_050.rb
require_relative 'shared'
require 'syslog'

# vuln-code-snippet start ruby_loginj_syslog_sprintf
def log_syslog_sprintf(req)
  Syslog.open('app', Syslog::LOG_PID, Syslog::LOG_USER)
  Syslog.log(Syslog::LOG_INFO, '%s', req.param('msg').gsub(/[\r\n]/, '')) # vuln-code-snippet safe-line ruby_loginj_syslog_sprintf
  Syslog.close
  BenchmarkResponse.json({ ok: true })
end
# vuln-code-snippet end ruby_loginj_syslog_sprintf

File: rust/testcode/loginjection_050.rs
//! CWE-117: User input stored in HashMap but safe constant value read and logged instead.

use std::collections::HashMap;

// vuln-code-snippet start testcodeLoginjection050
pub fn handle(req: &super::shared::BenchmarkRequest) -> super::shared::BenchmarkResponse {
    let mut entries = HashMap::new();
    entries.insert("user_input", req.param("username"));
    entries.insert("safe_msg", "login_event".to_string());
    let msg = entries.get("safe_msg").unwrap();
    log_info(&format!("event={}", msg)); // vuln-code-snippet target-line testcodeLoginjection050
    super::shared::BenchmarkResponse::ok("OK")
}

fn log_info(msg: &str) {
    eprintln!("[INFO] {}", msg);
}
// vuln-code-snippet end testcodeLoginjection050
